File: pinkmoe_server/model/request/follow.go
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-06-22 11:13:07
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-07 08:53:24
 * @FilePath: /pinkmoe_server/model/request/follow.go
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe 
 * 问题反馈qq群:749150798
 * xanaduCms程序上所有内容(包括但不限于 文字，图片，代码等)均为指针科技原创所有，采用请注意许可
 * 请遵循 “非商业用途” 协议。商业网站或未授权媒体不得复制内容，如需用于商业用途或者二开，请联系作者捐助任意金额即可，我们将保存所有权利。
 * Copyright (c) 2022 by 指针科技, All Rights Reserved.
 */
package request

import (
	"server/global"
	"server/model"
)

type ReqFollow struct {
	global.XD_MODEL
	FormUid         string       `json:"formUid" form:"formUid"` // 评论作者
	ToUid           string       `json:"toUid" form:"toUid"`     // 评论目标用户
	FormUidRelation model.XdUser `json:"formUidRelation" form:"formUidRelation"`
	ToUidRelation   model.XdUser `json:"toUidRelation" form:"toUidRelation"`
}

// SearchFollowParams api分页条件查询及排序结构体
type SearchFollowParams struct {
	ReqFollow
	PageInfo
	OrderKey string `json:"orderKey"` // 排序
	Desc     bool   `json:"desc"`     // 排序方式:升序false(默认)|降序true
}

File: pinkmoe_index/src/hooks/author/follow.ts
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-07-22 18:13:01
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-19 17:48:52
 * @FilePath: /pinkmoe_vitesse/src/hooks/author/follow.ts
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe   (如需用于商业用途或者二开，请联系作者捐助任意金额即可)
 * QQ:2419857357;支付宝:13135986153
 * Copyright (c) 2022 by coderzhaolu, All Rights Reserved.
 */
import type { ResPage } from '/@/api/common/types'
import { getAuthorFollowList } from '/@/api/author'
import type { ReqAuthorFollow, ResFollow } from '/@/api/author/types'

export const useAuthorFollow = () => {
  const authorFollowList = ref<ResPage<Array<ResFollow>>>()
  const route = useRoute()
  const loading = ref<boolean>(false)
  const hasMore = ref<boolean>(true)
  const formParams: ReqAuthorFollow = reactive({
    author: '',
    page: 1,
    pageSize: 12,
    orderKey: 'updated_at',
  })

  const nextPage = async () => {
    loading.value = true;
    (formParams.page as number)++
    const res = await getAuthorFollowList(formParams)
    if (!res.list || res.list.length <= 0)
      hasMore.value = false

    else
      authorFollowList.value?.list?.push(...res.list!)

    setTimeout(() => {
      loading.value = false
    }, 300)
  }

  // 获取关注列表
  const getAuthorFollow = async () => {
    loading.value = true
    formParams.formUid = route.params.author
    authorFollowList.value = await getAuthorFollowList(formParams)
    setTimeout(() => {
      loading.value = false
    }, 300)
  }

  onMounted(() => {
    getAuthorFollow()
  })

  return {
    authorFollowList,
    loading,
    hasMore,
    formParams,
    nextPage,
  }
}
